### PR TITLE
[Docs] Model_doc structure/clarity improvements

### PR DESCRIPTION
**Pull request by MKhalusova**
This PR streamlines pages in model_doc, so that a user who has seen at least one of these pages, should find it easier to navigate throughout any other model_doc page. These are the changes:

1. Individual pages now have (with a few exceptions) the same structure: overview, followed by author acknowledgement, usage tips and/or examples, resources (other docs, notebooks, blog posts, etc.), API reference (autodoc). Previously, some pages did not have the same order which could make searching through them confusing.
2. Usage tips now have their own section with a header. Previously, they often have been buried within the overview section, now one can find them faster.
3. In cases where a model has the same architecture as another model, a visible `<Tip>` has been added with a link to API reference of the original model. 
4. “Documentation resources” and “resources” were combined into “Resources”
5. For models that support TF and JAX, a `<framework>` tag was added so that a user could collapse sections irrelevant to them.
6. Removed disclaimers like “This model has been recently added. If you see something strange, file a GitHub Issue” for models that have been in the library longer than 1 year.

The PR aims to:
- Make it easier to find usage tips and examples
- Make it easier to find API reference for models that are based on another model’s architecture
- Possibly reduce the bloat by letting the users collapse framework-specific sections
- Streamline the model_doc pages structure

### CODE REVIEW

**Review by stevhliu — APPROVED** [review]
Wow really nice! 🎉  I love the idea of standardizing the pages so it's more consistent when going from model page to model page.

**Comment by MKhalusova** [comment]
If @ArthurZucker is busy, maybe @amyeroberts could take a look?

File: docs/source/en/model_doc/dinat.md [hunk 1]
```diff
@@ -76,6 +76,12 @@ A list of official Hugging Face and community (indicated by 🌎) resources to h
 
 If you're interested in submitting a resource to be included here, please feel free to open a Pull Request and we'll review it! The resource should ideally demonstrate something new instead of duplicating an existing resource.
 
+<Tip>
```
Comment (amyeroberts) [inline docs/source/en/model_doc/dinat.md:79]: Same here

File: docs/source/en/model_doc/dinov2.md [hunk 2]
```diff
@@ -22,13 +22,13 @@ The abstract from the paper is the following:
...
 
-Tips:
-
-- One can use [`AutoImageProcessor`] class to prepare images for the model.
-
 This model was contributed by [nielsr](https://huggingface.co/nielsr).
 The original code can be found [here](https://github.com/facebookresearch/dinov2).
 
+<Tip>
```
Comment (amyeroberts) [inline docs/source/en/model_doc/dinov2.md:28]: And here

**Review by amyeroberts — APPROVED** [review]
Mammoth piece of work! Thanks for all of this effort in making our documentation better ❤️ 

Main comment is about the tips for vision models mentioning the image processors. I realise this isn't something you added (I've approved in PRs) - it's just now in the refactoring and moving to tips I see it's not needed.

File: docs/source/en/model_doc/timesformer.md [hunk 3]
```diff
@@ -25,14 +25,13 @@ The abstract from the paper is the following:
...
 
-Tips:
-
-There are many pretrained variants. Select your pretrained model based on the dataset it is trained on. Moreover, the number of input frames per clip changes based on the model size so you should consider this parameter while selecting your pretrained model.
+There are many pretrained variants. Select your pretrained model based on the dataset it is trained on. Moreover,
```
[the remark this answers is not in the document]
Comment (MKhalusova) [reply docs/source/en/model_doc/timesformer.md:28]: Thanks for noticing! I missed this one.

**Pull request merged**